File: programmers/행렬의 곱셈.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer;
    int start_num=0;
    while(start_num!=arr1.size()){
        int start_num2=0;
        vector<int> sum;
        while(start_num2!=arr2[0].size()){
            long long num=0;
            for(int i=0;i<arr2.size();++i){
                num+=arr1[start_num][i]*arr2[i][start_num2];
            }
            sum.push_back(num);
            start_num2++;
        }
        answer.push_back(sum);
        start_num++;
    }
    return answer;
}

File: programmers/Level 2/행렬의 곱셈.js
function solution(arr1, arr2) {
    var answer = [[]];
    let start_num=0;
    answer.splice(0, answer.length);
    while(start_num!==arr1.length){
        let start_num2=0;
        let ans = [];
        while(start_num2!==arr2[0].length){
            let sum=0;
            for(let col=0;col<arr2.length;++col){
                sum+=(arr1[start_num][col]*arr2[col][start_num2]);
            }
            ans.push(sum);
            start_num2++;
        }
        answer.push(ans);
        start_num++;
    }
    
    return answer;
}
